### PR TITLE
use default constructors and skip unnecessary initialization

### PR DESCRIPTION
**Pull request by dagar**
Saves ~ 7.7KB of flash in px4fmu-v3_default.

### CODE REVIEW

**Comment by dagar** [comment]
We might also want to consider making the various constructors explicit to avoid unexpected conversions.

File: matrix/Matrix.hpp [hunk 1]
```diff
@@ -35,12 +35,12 @@ class Matrix
     // Constructors
     Matrix() = default;
 
-    Matrix(const Type data_[][N])
+    Matrix(const Type data_[M])
```
Comment (dagar) [inline matrix/Matrix.hpp]: TODO review
 - Mx1 constructor (vector)
 - do we need to support MxN matrix construction from an array that's M*N elements

**Comment by dagar** [comment]
Other than saving a surprising amount of flash I'm working towards closing all the memory safety holes.

File: matrix/Matrix.hpp [hunk 2]
```diff
@@ -24,28 +24,28 @@ namespace matrix
...
 
-    Type _data[M][N];
+    Type _data[M][N] {};
 
     // Constructors
-    Matrix() : _data() {}
+    Matrix() = default;
 
-    Matrix(const Type data_[][N]) : _data()
+    Matrix(const Type data_[M])
```
Comment (bkueng) [inline matrix/Matrix.hpp]: Do we need this one? If N!=1 you'll have an illegal memory access. Changing to `sizeof(data_)` would be fine too.

File: matrix/Scalar.hpp [hunk 3]
```diff
@@ -17,21 +17,18 @@ template<typename Type>
 class Scalar
 {
 public:
-    Scalar() : _value()
-    {
-    }
+    Scalar() = default;
```
Comment (bkueng) [inline matrix/Scalar.hpp]: Is this useful?

Comment (dagar) [reply matrix/Scalar.hpp]: Nope, originally wasn't sure if we could get away with Scalar being constant, but it appears that we can.

File: matrix/Matrix.hpp (hunk 2 above)
Comment (dagar) [reply matrix/Matrix.hpp]: We do for now, a lot of the testing constructs a matrix from an array. Check the updated commit.

**Pull request merged**